historical_system_profiles/db_interface.py

```diff
@@ -36,3 +37,15 @@ def get_hsps_by_profile_ids(profile_ids, account_number):
 
     query_results = query.all()
     return query_results
+
+
+def clean_expired_records(days_til_expired):
+    now = datetime.now()
+    expired_time = now - timedelta(days=days_til_expired)
+
+    query = HistoricalSystemProfile.query.filter(
+        HistoricalSystemProfile.created_on < expired_time
+    )
+    count = query.delete(synchronize_session="fetch")
+    db.session.commit()
+    return count
```

Comment:
Can be a follow up, with cards to track it:

 * Should we surface any metrics around number deleted when the loop runs?
 * Is there any debug data if the delete fails worth bubbling up to the exception log to track down issues?

Reply:
it should bubble up into the log but then keep going. For example, if I set the DB hostname wrong (lots of preceding stack trace cut out):

```
sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) could not translate host name "localhosst" to address: Name or service not known

(Background on this error at: http://sqlalche.me/e/e3q8)
MainThread INFO historical-system-profiles - sleeping for 20.0 minutes
```

Reply:
card created w/ metrics